### PR TITLE
add pre-commit

### PR DESCRIPTION
**Pull request by rhubert**
Add initial pre-commit configuration files. The pre-commit tool can be
    installed using pip:
       pip3 install pre-commit
    
    After that the hook can be activated using:
       pre-commit install
    
    or just run on the modified files
       pre-commit run
    
    ATM the following checks are active:
      - check-yaml
      - end-of-file-fixer
      - trailing-whitespace
      - yamllint
      - codespell

### CODE REVIEW

**Comment by jkloetzke** [comment]
Thanks. I don't think the `pyproject.toml` file is necessary. I pushed an update that moves it to `.codespellrc`. I hope that's ok.

**Comment by rhubert** [comment]
Thanks. :+1:

I just found out that `regex_exclude` setting of `yamlfmt` is somehow not what I want as it excludes the file if the regex matches. Removes all whitespaces (before comments, between `key:   value` pairs,...) making the yaml files look mostly ugly. :( Since the configuration options are very limited I think we won't use `yamlfmt`... Maybe `yamllint` to enable some checks + fix them by hand?

**Comment by jkloetzke** [comment]
> Maybe `yamllint` to enable some checks + fix them by hand?

Sounds reasonable. I guess we can fix the findings incrementally. The pre-commit thingy seems to support only looking at changed files.

File: .yamllint
Comment (jkloetzke) [inline .yamllint:1]: As this file is checked too, the indentation should probably be correct upfront.

File: .yamllint [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    min-spaces-from-content: 1
+  comments-indentation: enable
+  document-end: disable
+  document-start: disable
+  empty-lines:
+    max: 1
+    max-start: 0
+    max-end: 0
+  empty-values: disable
+  float-values: enable
```
Comment (jkloetzke) [inline .yamllint:40]: I think we can disable float values. Bob does not use them. Or do you use some plugin that uses float values?

File: .yamllint [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  anchors:
+    forbid-undeclared-aliases: true
+    forbid-duplicated-anchors: true
+    forbid-unused-anchors: true
+  braces:
+    forbid: false
+    min-spaces-inside: -1
+    max-spaces-inside: -1
+    min-spaces-inside-empty: -1
+    max-spaces-inside-empty: -1
```
Comment (jkloetzke) [inline .yamllint:17]: I would suggest:

```yaml
  braces:
    forbid: non-empty
```

Braces are not used in recipes. The only exception is `recipes/devel/cross-toolchain.yaml:73` where it's required.

**Review by jkloetzke — COMMENTED** [review]
Yeah, yamllint seems to be much more useful.

**Comment by jkloetzke** [comment]
Thanks! I'll try to add the check to the CI build in the next days.

**Pull request merged**